client/client-multi/client-objective-c/Kaa/DefaultBootstrapManager.m

```diff
@@ -0,0 +1,232 @@
+            NSMutableArray *servers = [self getTransportsByAccessPointId:[self.serverToApply intValue]];
+            if (servers && [servers count] > 0) {
+                [self notifyChannelManagerAboutServers:servers];
+                self.serverToApply = nil;
+            }
+        } else {
+            for (NSEnumerator *value in self.mappedIterators.allValues) {
+                id<TransportConnectionInfo> info = [[GenericTransportInfo alloc] initWithServerType:SERVER_OPERATIONS andMeta:[value nextObject]];
+                [self.channelManager onTransportConnectionInfoUpdated:info];
+            }
+        }
+    }
+}
+
+- (void)notifyChannelManagerAboutServers:(NSMutableArray *)servers {
+    for (ProtocolMetaData *meta in servers) {
+        DDLogDebug(@"%@ Applying new transport %@", TAG, meta);
+        GenericTransportInfo *info = [[GenericTransportInfo alloc] initWithServerType:SERVER_OPERATIONS andMeta:meta];
+        [self.channelManager onTransportConnectionInfoUpdated:info];
+    }
+}
+
+- (NSMutableArray *)getTransportsByAccessPointId:(int)accessPointId {
+    if (!self.operationsServerList || [self.operationsServerList count] <= 0) {
+        [NSException raise:KaaBootstrapRuntimeException format:@"Operations Server list is empty"];
+    }
+    NSMutableArray *result = [NSMutableArray array];
+    for (ProtocolMetaData *meta in self.operationsServerList) {
+        if (meta.accessPointId == accessPointId) {
+            [result addObject:meta];
+        }
+    }
+    return result;
+}
+
+- (void)applyDecision:(FailoverDecision *)decision {
+    switch (decision.failoverAction) {
+        case FAILOVER_ACTION_NOOP:
+        {
+            DDLogWarn(@"%@ No operation is performed according to failover strategy decision", TAG);
+        }
+            break;
+        case FAILOVER_ACTION_RETRY:
+        {
+            DDLogWarn(@"%@ Will try to receive operation servers in %li ms as to failover strategy decision",
+                      TAG, (long)(decision.retryPeriod));
+            __weak typeof(self)weakSelf = self;
+            dispatch_time_t delay = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(decision.retryPeriod * NSEC_PER_MSEC));
+            dispatch_after(delay, [self.context getSheduledExecutor], ^{
+                @try {
+                    [weakSelf receiveOperationsServerList];
+                }
+                @catch (NSException *exception) {
+                    DDLogWarn(@"%@ Excpetion caugh with name: %@, reason: %@", TAG, exception.name, exception.reason);
+                    DDLogError(@"%@ Error while receiving operations server list", TAG);
+                }
+            });
+        }
+            break;
+        case FAILOVER_ACTION_USE_NEXT_BOOTSTRAP:
+        {
+            DDLogWarn(@"%@ Trying to switch to the next bootstrap server as to failover strategy decision", TAG);
+            [self.failoverManager onServerFailed:[self.channelManager getActiveServer:TRANSPORT_TYPE_BOOTSTRAP]];
```

Comment:
Could this cause a force re-sync disregarding what a sync should be scheduled to execute after some time?